flags.go

```diff
@@ -116,38 +188,126 @@ func (f *Flags) setupCommonFlags() {
 }
+
 func (f *Flags) handleMaintenanceCommand() bool {
-	f.amtActivateCommand.StringVar(&f.Password, "password", f.lookupEnvOrString("AMT_PASSWORD", ""), "AMT password")
-	f.amtMaintenanceCommand.BoolVar(&f.SyncClock, "c", false, "sync AMT clock")
 
+	//validation section
 	if len(f.commandLineArgs) == 2 {
-		f.amtMaintenanceCommand.PrintDefaults()
+		f.printMaintenanceUsage()
 		return false
 	}
-	if err := f.amtMaintenanceCommand.Parse(f.commandLineArgs[2:]); err != nil {
+
+	task := ""
+	switch f.commandLineArgs[2] {
+	case "syncclock":
+		task = f.handleMaintenanceSyncClock()
+	case "syncip":
+		task = f.handleMaintenanceSyncIP()
+	case "changepassword":
+		task = f.handleMaintenanceSyncChangePassword()
+	default:
+		f.printMaintenanceUsage()
+	}
+	if task == "" {
 		return false
 	}
-	if f.amtMaintenanceCommand.Parsed() {
-		if f.URL == "" {
-			fmt.Println("-u flag is required and cannot be empty")
-			f.amtActivateCommand.Usage()
+	if f.URL == "" {
+		fmt.Print("\n-u flag is required and cannot be empty\n\n")
+		f.amtMaintenanceCommand.Usage()
+		return false
+	}
+	if f.Password == "" {
+		fmt.Println("Please enter the current AMT Password: ")
+		_, err := fmt.Scanln(&f.Password)
+		if f.Password == "" || err != nil {
+			fmt.Print("\ncurrent AMT password is required and cannot be empty\n\n")
+			f.amtMaintenanceCommand.Usage()
 			return false
 		}
-		if f.Password == "" {
-			fmt.Println("Please enter AMT Password: ")
-			var password string
-			// Taking input from user
-			_, err := fmt.Scanln(&password)
-			if password == "" || err != nil {
-				return false
+	}
+
+	f.Command = fmt.Sprintf("maintenance --password %s %s", f.Password, task)
+	return true
+}
+
+func (f *Flags) handleMaintenanceSyncClock() string {
+	if err := f.amtMaintenanceSyncClockCommand.Parse(f.commandLineArgs[3:]); err != nil {
+		return ""
+	}
+	return "--synctime"
+}
+
+func (f *Flags) handleMaintenanceSyncIP() string {
+	f.amtMaintenanceSyncIPCommand.Var(makeIpVar(&f.IpConfiguration.IpAddress), "staticip", "ip address for AMT")
+	f.amtMaintenanceSyncIPCommand.Var(makeIpVar(&f.IpConfiguration.Netmask), "netmask", "network mask")
+	f.amtMaintenanceSyncIPCommand.Var(makeIpVar(&f.IpConfiguration.Gateway), "gateway", "gateway address for AMT")
+	f.amtMaintenanceSyncIPCommand.Var(makeIpVar(&f.IpConfiguration.PrimaryDns), "primarydns", "primary domain name server")
+	f.amtMaintenanceSyncIPCommand.Var(makeIpVar(&f.IpConfiguration.SecondaryDns), "secondarydns", "secondary domain name server")
+
+	if err := f.amtMaintenanceSyncIPCommand.Parse(f.commandLineArgs[3:]); err != nil {
+		return ""
+	}
+	amtLanIfc, err := f.amtCommand.GetLANInterfaceSettings(false)
+	if err != nil {
+		log.Error(err)
+		return ""
+	}
+
+	ifaces, err := f.netEnumerator.Interfaces()
+	if err != nil {
+		log.Error(err)
+		return ""
+	}
+
+	for _, i := range ifaces {
+		// this handles the check for command line param and the first matched lookup value
+		if len(f.IpConfiguration.IpAddress) != 0 {
```

Comment:
If they've provided an IP address, I think skip getting the lan interface settings entirely and return up at line 253.

Reply:
done